src/app/course/course-item/highlight.directive.ts

```diff
@@ -0,0 +1,21 @@
+import { Directive, Input, ElementRef, OnInit } from "@angular/core";
+import { Course } from "../_models/course";
+import { daysdifference } from "src/app/shared-utils/date-time.utils";
+
+@Directive({ selector: "[appHighlight]" })
+export class HighlightDirective {
+    constructor(private el: ElementRef) {
+        this.el.nativeElement.style.border = "0.1rem solid";
+    }
+    @Input() public set course(course: Course) {
```

Comment:
please, declare Inputs before the constructor
And it would be nice to have some comments for publics